File: submissions/1098-largest-unique-number/solution.ts
function largestUniqueNumber(nums: number[]): number {
  const map = new Map();
  let max:number = 0;  
  let passed:boolean = true;  
  for(let i = 0; i < nums.length; i++){
      map.set(nums[i], (map.get(nums[i]) || 0) + 1);
  }  
    
   for(const [key,value] of map){
       if(value > 1){
           passed = false;
       } else {
           passed = true;
           max = Math.max(max, key)
       }
   } 
    if(passed) {
      return max;
    }
    
return -1;
};

File: submissions/1098-largest-unique-number/solution.py
class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        counts = defaultdict(int)
        ans = -1
        
        for num in nums:
            counts[num] += 1
        
        for num, appearance in counts.items():
            if appearance == 1:
                ans = max(ans, num)
                
        
        return ans
